### PR TITLE
SNOW-852382: Run Windows CI build on azure and gcp

### PR DESCRIPTION
**Pull request by sfc-gh-dprzybysz**
### Description
Let's Windows CI test scripts do not ignore CLOUD_PROVIDER env variable

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [x] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: ci/test_windows.bat [hunk 1]
```diff
@@ -8,7 +8,16 @@ call venv\scripts\activate
 pip install -U snowflake-connector-python
 
 cd %GITHUB_WORKSPACE%
-gpg --quiet --batch --yes --decrypt --passphrase=%PARAMETERS_SECRET% --output parameters.json .github/workflows/parameters_aws.json.gpg
+
+if "%CLOUD_PROVIDER%"=="AZURE" (
+  set ENCODED_PARAMETERS_FILE=.github/workflows/parameters_azure.json.gpg
+) else if "%CLOUD_PROVIDER%"=="GCP" (
+  set ENCODED_PARAMETERS_FILE=.github/workflows/parameters_gcp.json.gpg
+) else (
```
Comment (sfc-gh-igarish) [inline ci/test_windows.bat:16]: Nit: Right now we only support 3 providers, so last one implicitly become AWS. But way to check for AWS and use the aws json file and in final else fail the build with "unknown provider".

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Review by sfc-gh-pmotacki — APPROVED** [review]

File: ci/test_windows.bat (hunk 1 above)
Comment (sfc-gh-dprzybysz) [reply ci/test_windows.bat]: I added explicit check for AWS for Windows and also the same logic for Linux and Mac

**Comment by sfc-gh-pmotacki** [comment]
LGTM

**Pull request merged**